From 2a5d1519c300405845f0182fcfdd773c0a43c81a Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Tue, 22 Jan 2013 22:13:22 +0800
Subject: [PORT FROM MAIN]libstagefright: If nativewindow equals null, do not set scale mode

BZ: 81078

if nativewindow == NULL,native_window_set_scaling_mode will cause
TOMBSTONE error, add protection before native_window_set_scaling_mode

Change-Id: I2b3af0a431d27baf22ce218fde5de9668a716182
Orig-Change-Id: Id7d7dcec91f55aff1bb2bc9c6f4a46bbc3a19feb
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 media/libstagefright/ACodec.cpp |    9 +++++----
 1 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/media/libstagefright/ACodec.cpp b/media/libstagefright/ACodec.cpp
index 90bd4ce..eabe2e5 100644
--- a/media/libstagefright/ACodec.cpp
+++ b/media/libstagefright/ACodec.cpp
@@ -3451,10 +3451,11 @@ bool ACodec::LoadedState::onConfigureComponent(
                 static_cast<NativeWindowWrapper *>(obj.get()));
         CHECK(nativeWindow != NULL);
         mCodec->mNativeWindow = nativeWindow->getNativeWindow();
-
-        native_window_set_scaling_mode(
-                mCodec->mNativeWindow.get(),
-                NATIVE_WINDOW_SCALING_MODE_SCALE_TO_WINDOW);
+        if (mCodec->mNativeWindow != NULL) {
+            native_window_set_scaling_mode(
+                    mCodec->mNativeWindow.get(),
+                    NATIVE_WINDOW_SCALING_MODE_SCALE_TO_WINDOW);
+        }
     }
     CHECK_EQ((status_t)OK, mCodec->initNativeWindow());
 
-- 
1.7.4.1

